src/main/java/com/mraof/minestuck/entity/item/ItemRenderedProjectileEntity.java

```diff
@@ -0,0 +1,80 @@
+package com.mraof.minestuck.entity.item;
+
+import com.mraof.minestuck.client.renderer.entity.RendersAsItem;
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.IPacket;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+public class ItemRenderedProjectileEntity extends ProjectileItemEntity implements RendersAsItem
+{
+	public ItemRenderedProjectileEntity(EntityType<? extends ItemRenderedProjectileEntity> type, World worldIn)
+	{
+		super(type, worldIn);
+	}
+	
+	public ItemRenderedProjectileEntity(EntityType<? extends ItemRenderedProjectileEntity> type, double x, double y, double z, World worldIn)
+	{
+		super(type, x, y, z, worldIn);
+	}
+	
+	public ItemRenderedProjectileEntity(EntityType<? extends ItemRenderedProjectileEntity> type, LivingEntity livingEntityIn, World worldIn)
+	{
+		super(type, livingEntityIn, worldIn);
+	}
+	
+	@Override
+	public void writeAdditional(CompoundNBT compound)
+	{
+		super.writeAdditional(compound);
+	}
+	
+	@Override
+	public void readAdditional(CompoundNBT compound)
+	{
+		super.readAdditional(compound);
+	}
+	
+	@Override
+	protected void onImpact(RayTraceResult result)
+	{
+		if(this.getThrower() instanceof PlayerEntity)
```

Comment:
Why should everything in here only be executed if there is a thrower which is a player?

Reply:
I think I had gotten some sort of issue with crashes, but I have since made some overhauls through the addition of another entity class so let me send that for review